File: irisk-openapi-php-demo/report/RiskReportDataDemo.php
<?php
/** 本接口用于上报应用的举报信息。 */
/** 产品ID，从【易盾官网-服务管理-已开通业务】页面获取 */
const SECRET_ID = "your_secret_id";
/** 密钥，从【易盾官网-服务管理-已开通业务】页面获取 */
const SECRET_KEY = "your_secret_key";
const BUSINESS_ID = "your_business_id";
const VERSION = "500";
/** 接口URL */
const API_URL = "http://ir-open.dun.163.com/v5/risk/reportData";
/** API timeout*/
const API_TIMEOUT = 5;
require("../util.php");

/**
 * 反外挂请求接口简单封装
 * $params 请求参数
 */
function check($params)
{
    $params["secretId"] = SECRET_ID;
    $params["businessId"] = BUSINESS_ID;
    $params["version"] = VERSION;
    $params["timestamp"] = time() * 1000;
    // 随机码 32位
    $params["nonce"] = "mmm888f73yyy59440583zzz9bfcc79de";
    // 举报渠道
    $params['reportChannel'] = 'reportChannel';
    // 举报时间
    $params['reportTime'] = round(microtime(true) * 1000);
    // 举报人信息
    $whistleblowerMap = array(
        'account' => 'account01',
        'roleId' => 'roleId01',
        'roleName' => 'roleName01',
        'serverId' => 'serverId01',
        'level' => '10',
        'recharge' => '9999'
    );
    $params['whistleblower'] = json_encode($whistleblowerMap);
    // 被举报人信息
    $reportedPerson = array(
        'account' => 'account02',
        'roleId' => 'roleId02',
        'roleName' => 'roleName02',
        'serverId' => 'serverId02',
        'level' => '10',
        'recharge' => '9999'
    );
    $params['reportedPerson'] = json_encode($reportedPerson);
    // 举报类型
    $params['reportType'] = 'reportType01';
    // 举报场景
    $params['reportScene'] = 'reportScene01';
    // 举报内容详情
    $params['reportData'] = 'data';
    $params["signature"] = gen_signature(SECRET_KEY, $params);
    $params = toUtf8($params);


    $result = curl_post($params, API_URL, API_TIMEOUT);
    if ($result === FALSE) {
        return array("code" => 500, "msg" => "file_get_contents failed.");
    } else {
        return json_decode($result, true);
    }
}

// 简单测试
function main()
{
    echo "mb_internal_encoding=" . mb_internal_encoding() . "\n";

    $params = array(
    );

    $ret = check($params);
    var_dump($params);
    var_dump($ret);
    if ($ret["code"] == 200) {
        $data = $ret["data"];
        echo "请求成功，data: {$data}";
        var_dump($ret);
    } else {
        $msg = $ret["msg"];
        echo "请求失败，msg: {$msg}";
        var_dump($ret);
    }
}

main();
?>

File: irisk-openapi-go-demo/v5/report/RiskReportDataDemo.go
package main

import (
	"crypto/md5"
	"encoding/hex"
	"encoding/json"
	"fmt"
	simplejson "github.com/bitly/go-simplejson"
	"io/ioutil"
	"net/http"
	"sort"
	"strconv"
	"strings"
	"time"
)

/**
 * 本接口用于上报应用的举报信息。
 */

const (
	apiUrl     = "http://ir-open.dun.163.com/v5/risk/reportData" //接口url
	secretId   = "your_secret_id"                                //产品id，每个应用接入时，会分配secretId和私钥secretKey。
	secretKey  = "your_secret_key"                               //产品密钥，每个应用接入时，会分配secretId和私钥secretKey。
	businessId = "your_business_id"                              //每个业务接入时，均会分配业务 ID，有对应的密钥 secretId。
	version    = "500"                                           //版本号，如500
)

// 请求易盾接口
func check(params map[string]interface{}) *simplejson.Json {
	marshal, _ := json.Marshal(params)
	jsonStr := string(marshal)
	println(jsonStr)
	resp, err := http.Post(apiUrl, "application/json", strings.NewReader(jsonStr))

	if err != nil {
		fmt.Println("调用API接口失败:", err)
		return nil
	}

	defer resp.Body.Close()

	contents, _ := ioutil.ReadAll(resp.Body)
	result, _ := simplejson.NewJson(contents)
	return result
}

// 生成签名信息
func genSignature(params map[string]interface{}) string {
	var paramStr string
	keys := make([]string, 0, len(params))
	for k := range params {
		keys = append(keys, k)
	}
	sort.Strings(keys)
	for _, key := range keys {
		paramStr += key + params[key].(string)
	}
	paramStr += secretKey
	md5Reader := md5.New()
	md5Reader.Write([]byte(paramStr))
	return hex.EncodeToString(md5Reader.Sum(nil))
}

func main() {
	params := make(map[string]interface{})
	// 调用接口当前时间，单位毫秒
	params["secretId"] = secretId
	params["businessId"] = businessId
	params["nonce"] = "mmm888f73yyy59440583zzz9bfcc79de"
	params["timestamp"] = strconv.FormatInt(time.Now().UnixNano()/1000000, 10)
	params["version"] = version

	// 举报渠道
	params["reportChannel"] = "reportChannel"
	// 举报时间
	params["reportTime"] = strconv.FormatInt(time.Now().UnixNano()/1000000, 10)
	// 举报人信息
	whistleblowerMap := make(map[string]interface{})
	whistleblowerMap["account"] = "account01"
	whistleblowerMap["roleId"] = "roleId01"
	whistleblowerMap["roleName"] = "roleName01"
	whistleblowerMap["serverId"] = "serverId01"
	whistleblowerMap["level"] = "10"
	whistleblowerMap["recharge"] = "9999"
	whistleblower, _ := json.Marshal(whistleblowerMap)
	params["whistleblower"] = string(whistleblower)
	// 被举报人信息
	reportedPerson := make(map[string]interface{})
	reportedPerson["account"] = "account02"
	reportedPerson["roleId"] = "roleId02"
	reportedPerson["roleName"] = "roleName02"
	reportedPerson["serverId"] = "serverId02"
	reportedPerson["level"] = "10"
	reportedPerson["recharge"] = "9999"
	reportedPersonJSON, _ := json.Marshal(reportedPerson)
	params["reportedPerson"] = string(reportedPersonJSON)
	// 举报类型
	params["reportType"] = "reportType01"
	// 举报场景
	params["reportScene"] = "reportScene01"
	// 举报内容详情
	params["reportData"] = "data"
	params["signature"] = genSignature(params)

	ret := check(params)

	code, _ := ret.Get("code").Int()
	message, _ := ret.Get("msg").String()
	data, _ := ret.Get("data").MarshalJSON()
	if code == 200 {
		fmt.Printf("请求成功：msg=%s, data=%s", message, data)
	} else {
		fmt.Printf("请求失败: code=%d, msg=%s", code, message)
	}
}
